### PR TITLE
Add CHANGELOG.md entry for 5.14.1

### PR DESCRIPTION
**Pull request by lucasmrod**
Generated the following way:
```
go run ./cmd/release-notes -github-token ... -changelog ../forks/osquery/CHANGELOG.md -last 5.13.1 -new 5.14.1 > NEW.txt
```

And then edited manually.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,57 @@
 # osquery Changelog
 
+<a name="5.14.1"></a>
+## [5.14.1](https://github.com/osquery/osquery/releases/tag/5.14.1)
+
+[Git Commits](https://github.com/osquery/osquery/compare/5.13.1...5.14.1)
+
+Representing commits from 13 contributors! Thank you all.
+
+### New Features
```
Comment (directionless) [inline CHANGELOG.md:10]:
```suggestion
Representing commits from 13 contributors! Thank you all.

### Windows codesigning note

Starting with Osquery 5.14, we have changed our codesigning. Henceforth our releases will be signed by an osquery specific signing key issues by Microsoft Azure. 

### New Features
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+
+- Add `--yara_sigurl_authenticate` flag ([#8437](https://github.com/osquery/osquery/pull/8437))
+
+### Table Changes
+
+- Add additional WMI data to `deviceguard_status` table ([#8440](https://github.com/osquery/osquery/pull/8440))
+- Fix linux `groups` table to handle larger group sets by increasing buffer size ([#8387](https://github.com/osquery/osquery/pull/8387))
+- Update table `alf_explicit_auths` as not supported on macOS 15 ([#8435](https://github.com/osquery/osquery/pull/8435))
+- Update table `alf_exceptions` to support macOS 15 ([#8434](https://github.com/osquery/osquery/pull/8434))
+- Add column optimization support to allow processing `IN` constraints all at once in xFilter ([#8263](https://github.com/osquery/osquery/pull/8263))
```
Comment (directionless) [inline CHANGELOG.md:20]: This is probably under the hood. It shouldn't have any table usage changes

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,10 @@
 
 Representing commits from 13 contributors! Thank you all.
 
+### Windows codesigning note
+
+Starting with Osquery 5.14, we have changed our codesigning. Henceforth our releases will be signed by an osquery specific signing key issues by Microsoft Azure.
```
Comment (Smjert) [inline CHANGELOG.md:12]:
```suggestion
Starting with Osquery 5.14, we have changed our codesigning. Henceforth our releases will be signed by an osquery specific signing key issued by Microsoft Azure. 
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,61 @@
...
+### New Features
+
+- Add `--yara_sigurl_authenticate` flag ([#8437](https://github.com/osquery/osquery/pull/8437))
+
+### Table Changes
+
+- Add additional WMI data to `deviceguard_status` table ([#8440](https://github.com/osquery/osquery/pull/8440))
+- Fix linux `groups` table to handle larger group sets by increasing buffer size ([#8387](https://github.com/osquery/osquery/pull/8387))
+- Add support for Firefox addons for snap installations ([#8374](https://github.com/osquery/osquery/pull/8374))
+- table: Remove support for deprecated Safari Legacy Extensions ([#8426](https://github.com/osquery/osquery/pull/8426))
```
Comment (directionless) [inline CHANGELOG.md:23]:
```suggestion
- Remove support for deprecated Safari Legacy Extensions ([#8426](https://github.com/osquery/osquery/pull/8426))
```

**Review by directionless — APPROVED** [review]

**Pull request merged**